src/EnergyPlus/VentilatedSlab.cc

```diff
@@ -1135,170 +1074,172 @@ namespace VentilatedSlab {
-                                  CurrentModuleObject,
-                                  state.dataVentilatedSlab->VentSlab(Item).Name + "-SYSTEM",
-                                  state.dataIPShortCut->cAlphaArgs(30),
-                                  state.dataIPShortCut->cAlphaArgs(31),
-                                  state.dataIPShortCut->cAlphaArgs(24),
-                                  "UNDEFINED");
-
-                    // Add heating coil to component sets array when cooling coil present
-                    SetUpCompSets(state,
-                                  CurrentModuleObject,
-                                  state.dataVentilatedSlab->VentSlab(Item).Name + "-SYSTEM",
-                                  state.dataIPShortCut->cAlphaArgs(27),
-                                  state.dataIPShortCut->cAlphaArgs(28),
-                                  "UNDEFINED",
-                                  state.dataIPShortCut->cAlphaArgs(19));
-
-                } else if (SELECT_CASE_var == HeatingOption) { // 'Heating'
-                    // Add heating coil to component sets array when no cooling coil present
-                    SetUpCompSets(state,
-                                  CurrentModuleObject,
-                                  state.dataVentilatedSlab->VentSlab(Item).Name + "-SYSTEM",
-                                  state.dataIPShortCut->cAlphaArgs(27),
-                                  state.dataIPShortCut->cAlphaArgs(28),
-                                  state.dataIPShortCut->cAlphaArgs(24),
-                                  state.dataIPShortCut->cAlphaArgs(19));
-
-                } else if (SELECT_CASE_var == CoolingOption) { // 'Cooling'
-                    // Add cooling coil to component sets array when no heating coil present
-                    SetUpCompSets(state,
-                                  CurrentModuleObject,
-                                  state.dataVentilatedSlab->VentSlab(Item).Name + "-SYSTEM",
-                                  state.dataIPShortCut->cAlphaArgs(30),
-                                  state.dataIPShortCut->cAlphaArgs(31),
-                                  state.dataIPShortCut->cAlphaArgs(24),
-                                  state.dataIPShortCut->cAlphaArgs(19));
-
-                } else if (SELECT_CASE_var == NoneOption) {
-
-                } else {
-                }
+            switch (ventSlab.coilOption) {
+            case CoilType::Both: { // 'HeatingAndCooling'
+                // Add cooling coil to component sets array when present
+                SetUpCompSets(state,
+                              CurrentModuleObject,
+                              ventSlab.Name + "-SYSTEM",
+                              state.dataIPShortCut->cAlphaArgs(30),
+                              state.dataIPShortCut->cAlphaArgs(31),
+                              state.dataIPShortCut->cAlphaArgs(24),
+                              "UNDEFINED");
```

Comment:
The use of cAlphaArgs(*) hides what these are and could cause problems in the future. 

                SetUpCompSets(state,
                              CurrentModuleObject,
                              ventSlab.Name + "-SYSTEM",
                              state.dataIPShortCut->cAlphaArgs(30), // Cooling Coil Object Type
                              state.dataIPShortCut->cAlphaArgs(31), // Cooling Coil Name
                              state.dataIPShortCut->cAlphaArgs(24), // Cooling Coil Inlet Node Name
                              "UNDEFINED");                                       // Cooling Coil Outlet Node Name

 I would rather see literal text that will get updated when model inputs change.

                SetUpCompSets(state,
                              CurrentModuleObject,
                              ventSlab.Name + "-SYSTEM",
                              state.dataIPShortCut->ventSlab.coolingCoilTypeCh,
                              state.dataIPShortCut->ventSlab.cCoilType,
                              state.dataIPShortCut->state.dataLoopNodes->NodeID(ventSlab.FanOutletNode),
                              "UNDEFINED");

Reply:
Yes, that is a good pattern. As soon as an IP argument is captured in an object field, just use that object field rather than the IP argument.

Reply:
Welll, this may have to stay this way until there's a refactoring of the `*CompSets` functions. We're moving away from saving the names in the struct, but `CompSets` still works off of string names, so the input alphas are still needed, for now.

Reply:
The ultimate solution is to use epJSON, which is name based.

Reply:
Welll, even there, you need a local string variable to hold the name for later use. So maybe that's the solution for now, if the name isn't being saved in the struct, then save it in a local variable for use in other function calls and error messages.